src/components/XdrViewer.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import _ from 'lodash';
+import SelectPicker from './FormComponents/SelectPicker';
+import extrapolateFromXdr from '../utilities/extrapolateFromXdr';
+import TreeView from './TreeView';
+import validateBase64 from '../utilities/validateBase64';
+import {updateXdrInput, updateXdrType} from '../actions/xdrViewer';
+import StellarSdk from 'stellar-sdk';
+
+function XdrViewer(props) {
+  let {dispatch, state} = props;
+
+  let validation = validateBase64(state.input);
+  let messageClass = validation.result === 'error' ? 'xdrInput__message__alert' : 'xdrInput__message__success';
+  let message = <p className={messageClass}>{validation.message}</p>
+
+  let xdrTypeIsValid = _.indexOf(xdrTypes, state.type) >= 0;
+  let treeView, errorMessage;
+  if (state.input === '') {
+    errorMessage = <p>Enter a base-64 encoded xdr blob to decode.</p>;
+  } else if (state.type === '' || !xdrTypeIsValid) {
```

Comment:
Good points.

I thought that the xdrTypes array had an empty string element. Turns out the `xdrTypes` array does not and that comparison to `''` would be caught by `xdrTypeIsValid`.

I do `xdrTypeIsValid` since the value is simply a get value in the hash url.